packages/components/src/popover/style.scss

```diff
@@ -60,12 +60,64 @@
 
 .components-popover__arrow {
 	position: absolute;
-	background: $gray-400;
-	width: 8px;
-	height: 8px;
-	transform: rotate(45deg);
-	z-index: -1;
-	.is-alternate & {
-		background: $gray-900;
+	width: var(--wp-popover-arrow-size);
+	height: var(--wp-popover-arrow-size);
+	pointer-events: none;
+
+	// Thin line that helps to make sure that the underlying
+	// popover__content's outline is fully overlapped by the
+	// arrow
+	&::before {
+		content: "";
+		position: absolute;
+		top: -1px;
+		left: 0;
+		height: 2px;
+		width: 100%;
+		background-color: $white;
+	}
+
+	// Position and rotate the arrow depending on the popover's placement.
+	// The `!important' is necessary to override the inline styles.
+	&--top {
+		bottom: calc(-1 * var(--wp-popover-arrow-size)) !important;
+		transform: rotate(0);
+	}
+	&--right {
+		/*rtl:begin:ignore*/
+		left: calc(-1 * var(--wp-popover-arrow-size)) !important;
+		transform: rotate(90deg);
+	}
+	&--bottom {
+		top: calc(-1 * var(--wp-popover-arrow-size)) !important;
+		transform: rotate(180deg);
+	}
+	&--left {
+		/*rtl:begin:ignore*/
+		right: calc(-1 * var(--wp-popover-arrow-size)) !important;
+		transform: rotate(-90deg);
+		/*rtl:end:ignore*/
+	}
+}
+
+.components-popover__triangle {
+	fill: transparent;
+	position: absolute;
+	height: 100%;
+	width: 100%;
+	stroke-width: $border-width;
+
+	// Same as popover content
+	&-bg {
+		fill: $white;
+		stroke: $white;
+	}
+
+	&-border {
```

Comment:
Nit: This might be a personal preference thing, but I prefer full class names to be written out (`.components-popover__triangle-bg`) so they can be searched easily 🥺

Reply:
Done in https://github.com/WordPress/gutenberg/pull/42874/commits/4a00b8dd36f33c0eff09669629e3ba29237db451